clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -9300,6 +9300,12 @@ def warn_unused_container_subscript_expr : Warning<
 def warn_unused_call : Warning<
   "ignoring return value of function declared with %0 attribute">,
   InGroup<UnusedValue>;
+def warn_unused_return_type : Warning<
+  "ignoring %select{return value|temporary}0 of type %2 declared with %1 attribute">,
+  InGroup<UnusedValue>;
+def warn_unused_return_type_msg : Warning<
```

Comment:
Rather than separate diagnostics, can we do this with a `%select` for the last argument?  So:

`ignoring %select{return value|temporary}0 of type %2 declared with %1 attribute%select{: %4}3`.

Reply:
I originally wanted to do this, but existing `warn_unused_constructor` and `warn_unused_result` all have a separate `_msg` version... If we fold together the two `unused_return_type`s, should these two be folded together too?

Reply:
I'll leave the choice up to you!  Change all of them would be my preference, but I'd like consistency between the three.

Reply:
Sure, I will collapse all three then.